packages/pg-cursor/index.js

```diff
@@ -133,7 +135,9 @@ class Cursor extends EventEmitter {
 
   handleCommandComplete(msg) {
     this._result.addCommandComplete(msg)
-    this._closePortal()
+    if (!this.moveAllAndClosing) {
+      this._closePortal()
+    }
```

Comment:
I remain convinced this is a bug: Simple queries can be run intermixed with extended queries (https://segmentfault.com/a/1190000017136059 points to that and we do it, infact) and the issuance of simple queries should not cause the portal to be closed, especially when it's just suspended.

For example, a user connects with postgres via node-postgres and issues the following wire protocol messages (pseudocode):
```
PARSE query "select generate_series(1,$1)" as prepared statement "foo" <- returns ParseComplete
BIND prepared statement "foo" to portal "bar" with parameters: (10) <- returns BindComplete
EXECUTE on portal bar (5) rows <- returns PortalSuspended
QUERY "select 1" <- returns CommandComplete and node-postgres closes the portal bar
EXECUTE on portal bar (5) rows <- ErrorMessage because bar is already closed
```

The fix is that once this cursor object receives a PortalSuspended, it should not pay any attention to future messages until it issues another extended query command. Per the spec, multiple queries in flight are allowed, the client is just responsible for maintaining a queue to manager responses correctly.

For our purposes this solution works! I'd prefer we build this unique code on top of the package rather than have it be in the package itself

Reply:
How would you solve the problem involving 'begin' statements?

In the original package, issuing a 'begin' command gives a CommandComplete. Without closing the portal inside handleCommandComplete, the server doesn't send anything else and the cursor will hang. No PortalSuspendeds are involved in this scenario. Not sure why this happens or if there's something deeper we can dig into

Reply:
I think a begin statement issued through a simple query should be handled just as the "select 1" query in the example above - once a `PortalSuspended|ErrorResponse|EmptyQueryResponse|CommandComplete` message has been received by the client after an Execute was called, the client should not trigger cursor behavior on subsequent messages from the server until the client issues more extended query protocol messages and then flushes/syncs. 

Another way of saying this is that every sent message has a paired response(s) [some commands generate multiple messages] from the server. In this case, the node-postgres cursor object considers any command complete response as being associated with it even if it isn't waiting on a paired response. In fact, I'm guessing we could make a cursor, not read any lines (i.e, not call execute), send a simple query, and the cursor will be closed. A step further, it's possible (in terms of the wire protocol) to have multiple cursors concurrently (parse + bind + execute). Not sure what the behavior of how the cursor handles commandcomplete is there, either.

Reply:
For the scope of this PR and our application, we can move beyond this point. As a follow up,

Can you create a small reproducible example on plain old node-postgres that:
1) creates a cursor for `select generate_series(1,1000)`
2) retrieves 100 rows
3) runs `select 1`
4) retrieves another 100 rows

Reply:
Made a small test case. The client hangs instead of erroring -- any code after the ```select 1``` isn't executed. The sequence is: PortalSuspended (from first cursor read), CommandComplete (from select 1), then a readyForQuery, CloseComplete, readyForQuery and no code is executed afterwards. In any case the cursor is closed prematurely

Reply:
Made a small test case -- the client's behavior seems okay, now that I think about it. The ```select 1``` only causes an issue if you query it directly from ```client.connection``` or ```cursor.connection```, which I think is something that people typically don't do. In our case it yields an error. However, if you query like ```client.query("select 1")```, then the client behaves as expected -- the query is not executed until the cursor is finished.

Reply:
Got it. I see now this is implemented with a queryQueue that waits on ReadyForQuery. So this behavior isn't a bug b/c one cannot send two queries (simple or extended) using the high-level API until you get all the results from the first one.  For another day is adding client-side support support for having multiple extended query portals :-)

Reply:
Can you post the example here or send it to me?

Reply:
Yup! this is code demonstrating that the portal doesn't close after calling client.query: can be modified to show that the portal closes after calling client.connection.query instead
```
const { Client } = require('pg')
const Cursor = require('pg-cursor')

const client = new Client({
    database: 'postgres',
    user: 'bitdotio-2',
    password: 'password',
    port: '5433',
})

client.connect()

const cursor = client.query(new Cursor("SELECT * FROM generate_series(1,100)"))
// read partially from cursor
cursor.read(1, () => {
    console.log("Cursor read")
    // send query from client, which will not execute since there is a PortalSuspended
    client.query("SELECT 1", () => {
        console.log("SELECT 1")
        cursor.read(100, () => {
            console.log("Cursor read again")
        })
    })
    // if calling client.connection.query() or cursor.connection.query()
    // instead, the portal will receive a commandComplete and then close
})
```